torch/jit/__init__.py

```diff
@@ -1460,25 +1464,19 @@ def __init__(self, original, stubs):
                     object.__setattr__(self, name, item)
                 elif isinstance(item, Parameter) or (isinstance(item, Module) and item is not self):
                     ScriptModule.__setattr__(self, name, item)
+
+            # Copy buffers
             for name in original._buffers:
                 if original._buffers[name] is None:
                     object.__setattr__(self, name, None)
                 else:
                     self.register_buffer(name, original._buffers[name])
 
             # Copy constants
-            self.__dict__["_constants_set"] = set(getattr(original, "__constants__", []))
-
-            # Automatically register primitive types as attributes
-            for name in dir(original):
-                if name == 'training':
-                    # TODO: remove this hack when training is a bool attribute
-                    # instead of a buffer
-                    continue
-                item = getattr(original, name)
-                primitive_type = torch.jit.annotations.get_primitive_type(item)
-                if name not in self.__dict__["_constants_set"] and primitive_type:
-                    ScriptModule.__setattr__(self, name, torch.jit.Attribute(item, primitive_type))
+            self.__dict__["_constants_set"] = constants_set
```

Comment:
Could you move the copy constants logic below the `self.__dict__["_constants_set"] = {}` line ? I think that will read more clearly.

Reply:
This line has to happen after the parameters due to #14715